prusti-server/src/process_verification.rs

```diff
@@ -8,12 +8,45 @@ use crate::{VerificationRequest, ViperBackendConfig};
 use log::info;
 use prusti_common::{config, report::log::report, vir::ToViper, Stopwatch};
 use std::{fs::create_dir_all, path::PathBuf};
-use viper::{VerificationBackend, VerificationContext};
+use viper::{Cache, VerificationBackend, VerificationContext};
+
+pub fn process_verification_request_cache<'v, 't: 'v>(
+    verification_context: &'v VerificationContext<'t>,
+    request: VerificationRequest,
+    cache: impl Cache,
+) -> viper::VerificationResult {
+    if config::disable_cache() || config::print_hash() {
+        process_verification_request(verification_context, request)
+    } else {
+        let hash = request.get_hash();
+        // Try to load from cache and return `result`
+        if let Some(result) = cache.get(hash) {
+            return result;
```

Comment:
```suggestion
            info!("Using verification result from the cache");
            return result;
```

Just to make debugging easier in the IDE.